app/src/composables/use-reload-guard.ts

```diff
@@ -0,0 +1,22 @@
+import { type Ref, ref } from 'vue';
+import { useRoute } from 'vue-router';
+import { useNavigationGuard } from './use-navigation-guard';
+
+export function useReloadGuard(needsReload: Ref<boolean>) {
+	const { path } = useRoute();
+
+	const confirmLeave = ref(false);
+	const leaveTo = ref<string | null>(null);
+
+	const { removeGuard } = useNavigationGuard(needsReload, (to) => {
+		if (needsReload.value && !to.path.startsWith(path)) {
+			confirmLeave.value = true;
+			leaveTo.value = to.fullPath;
+			return false;
+		}
+
+		return true;
+	});
+
+	return { confirmLeave, leaveTo, removeGuard };
+}
```

Comment:
`removeGuard` is not required, see comment in `extensions.vue`

```suggestion
import { ref, type Ref } from 'vue';
import { useRoute } from 'vue-router';
import { useNavigationGuard } from './use-navigation-guard';

export function useReloadGuard(needsReload: Ref<boolean>) {
	const { path } = useRoute();

	const confirmLeave = ref(false);
	const leaveTo = ref<string | null>(null);

	useNavigationGuard(needsReload, (to) => {
		if (needsReload.value && !to.path.startsWith(path)) {
			confirmLeave.value = true;
			leaveTo.value = to.fullPath;
			return false;
		}

		return true;
	});

	return { confirmLeave, leaveTo };
}
```

Reply:
Ah the combination with the vue-use `useEventListener` was the big trick here to get rid of those browser/system dialogs 🤔